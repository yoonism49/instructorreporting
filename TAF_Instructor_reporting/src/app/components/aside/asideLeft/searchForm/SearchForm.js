// @flow weak

import React, {
  PureComponent
}                     from 'react';
import PropTypes      from 'prop-types';

class SearchForm extends PureComponent {
  static propTypes = {
    onSearchSubmit: PropTypes.func
  };

  searchinput = null;

  componentDidMount() {
    this.searchinput.focus();
  }

  render() {
    return (
      <form
        className="sidebar-form"
        onKeyPress={this.handlesFormKeyPress}>
        <div className="input-group">
          <input
            ref={this.getRef}
            type="text"
            name="searchinput"
            className="form-control"
            placeholder="Search..."/>
          <span className="input-group-btn">
            <button
              onClick={this.handlesFormKeyPress}
              name="search"
              id="search-btn"
              className="btn btn-flat">
              <i className="fa fa-search"></i>
            </button>
          </span>
        </div>
      </form>
    );
  }

  getRef = (ref) => (this.searchinput = ref)

  handlesFormKeyPress = (event) => {
    if (event.charCode === 13) {
      event.preventDefault();
      const { onSearchSubmit } = this.props;
      onSearchSubmit(this.searchinput.value.trim());
    }
  }
}

export default SearchForm;
