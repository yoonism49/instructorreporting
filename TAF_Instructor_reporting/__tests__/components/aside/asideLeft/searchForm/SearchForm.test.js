'use strict';

import React          from 'react';
import SearchForm     from '../../../../../src/app/components/aside/asideLeft/searchForm/SearchForm';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { mount }      from 'enzyme';

describe('SearchForm component', () => {
  const onSubmitMock    = jest.fn();
  const preventDefault  = jest.fn();
  const ENTER_KEY       = 13;

  function createNodeMock() {
    // prevents error from this.searchinput.focus() 
    // (searchinput is ref to input)
    return {
      focus() {}
    };
  }

  it('renders as expected', () => {
    const component = renderer.create(
      <SearchForm
        isActive={onSubmitMock}
      />,
      { createNodeMock }
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should sumbit on enterKey press', () => {
    const mockEnterKeyPressEvent = {
      charCode: ENTER_KEY,
      preventDefault
    };
  
    const SearchFormComponent = mount(
      <SearchForm
        onSearchSubmit={onSubmitMock}
      />
    );

    SearchFormComponent
      .find('form')
      .simulate('keypress', mockEnterKeyPressEvent);

    expect(onSubmitMock).toHaveBeenCalled();
  });
});
