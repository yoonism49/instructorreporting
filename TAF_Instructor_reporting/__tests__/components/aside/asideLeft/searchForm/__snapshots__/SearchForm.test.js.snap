// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders as expected 1`] = `
<form
  className="sidebar-form"
  onKeyPress={[Function]}
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      name="searchinput"
      placeholder="Search..."
      type="text"
    />
    <span
      className="input-group-btn"
    >
      <button
        className="btn btn-flat"
        id="search-btn"
        name="search"
        onClick={[Function]}
      >
        <i
          className="fa fa-search"
        />
      </button>
    </span>
  </div>
</form>
`;
